### ISSUE TITLE
[Bug][A11Y][Pin Code Input] Role is missing on digit code input

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#486]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

S22

### OS Version

Android

### App version

0.6.0

### Describe the issue

On Android, the digit input has 2 focusable elements with Talk Back :
- the "parent" digit code with correct name ("digit code 1") but **no role**, 
- and the "child" text field inside the input which gives the role. 

<img width="379" height="618" alt="" src="https://github.com/user-attachments/assets/d48b7ab8-ac2b-4e65-9c7c-d31555ba4450" />

### Expected Behavior

Group elements together so there will be only one focusable element represented by the parent element with correct name and role

~~It's done correctly on iOS, we hear "Digit code 1, text field"~~

**update : on iOS 1.1.1** : same behavior, please group container and text field for each digit with the proper label

**Comment by pya35** [comment]
OK
